src/payment/strategies/sdk/hosted-form-strategy.ts

```diff
@@ -0,0 +1,97 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { HostedForm, HostedFormFactory } from '../../../hosted-form';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { handler } from './handler';
+
+/*
+    BIG TODO: How do we handle the return journey of redirects?
+*/
+
+/*
+    This strategy would be similar-ish to src/payment/strategies/credit-card/credit-card-payment-strategy.ts
+
+    Key points:
+        - HostedFormFactory would be either extended or cloned and modified to
+            (1) allow the post action to be constructed as needed
+                e.g. `/payment/${paymentProvider}.${methodId}/${paymentId}`
+            (2) transparently return the response from this submission
+                allowing it to be processed here, in the strategy scope
+
+        - The use of a "PPSDK Response Handler", which would
+            (1) discern different code paths based on success/failure/error/continue
+                responses from our checkout/processing endpoint (received via HostedFormFactory above)
+            (2) not return control to the strategy until all continue steps are exhausted
+            (3) attempt to handle any recoverable failure responses(?)
+            (4) translate error and un-recoverable failure responses into Checkout SDK errors
+                (e.g. those from src/common/error/errors/index.ts)
+                before throwing
+            (5) Delegate and await the resolving of relevant(?) continue steps to checkout-js
+            (6) Return a resolving promise from 'execute'
+                upon a success message from the checkout/processing endpoint
+
+*/
+
+export class HostedFormStrategy implements PaymentStrategy {
+    protected _hostedForm?: HostedForm;
+
+    constructor(
+        protected _store: CheckoutStore,
+        protected _orderActionCreator: OrderActionCreator,
+        protected _hostedFormFactory: HostedFormFactory
+    ) {}
+
+    initialize(options?: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
```

Comment:
How come for CC methods, we don't need to do `POST /payment/provider.method` to know we need to show a CC form. But we do with other methods, like redirect?

Reply:
Here we were making that decision based off the `paymentMethod.method === 'card'` check that had already occurred in order to choose this strategy `PaymentStrategyType.SDK_HOSTED`.

Might be a moot point after I try unifying the strategies, stay tuned 😄

Reply:
The non-hosted 'branch' has been effectively removed from this PR, hopefully making the comparison moot and this bit of conversation resolvable 😄